alkemio_virtual_contributor_engine/prompt_graph/node.py

```
@@ -0,0 +1,90 @@
+            else:
+                raise KeyError(
+                    f"Required input variable '{var}' not found in state "
+                    f"for node '{self.name}'"
+                )
+        try:
+            return self.prompt.format(**values)
+        except KeyError as e:
+            raise KeyError(
+                f"Prompt template for node '{self.name}' requires variable "
+                f"{e} which was not provided"
+            )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)